src/hooks/useDebounce.ts

```diff
@@ -0,0 +1,17 @@
+import { useState, useEffect } from 'react';
+
+export function useDebounce<T>(value: T, delay: number): T {
+  const [debouncedValue, setDebouncedValue] = useState<T>(value);
+
+  useEffect(() => {
+    const handler = setTimeout(() => {
+      setDebouncedValue(value);
+    }, delay);
+
+    return () => {
+      clearTimeout(handler);
```

Comment:
will this clear timeout if you continue typing before the delay is complete?

Reply:
Yes, this will clear the timeout if you continue typing before the delay is complete. It ensures that the debounced value only updates after you've stopped typing for the specified delay period. Each new keystroke cancels the previous pending update and starts a new timer.

Reply:
❤️ awesome! Can we get rid of the other debounce that we have / use when we search, maybe we  can replace it with this? I was using `rxjs` debounce, maybe we can use that or if not, we can maybe remove rxjs entirely since I don't think we are using it for anything else... I was thinking that adjusting to use observables may be beneficial, but if we are not using it, then we should remove